sklearn/utils/estimator_checks.py

```diff
@@ -424,6 +424,7 @@ def _maybe_mark(
     expected_failed_checks: dict[str, str] | None = None,
     mark: Literal["xfail", "skip", None] = None,
     pytest=None,
+    strict: bool | None = None,
```

Comment:
Maybe it would be a bit clearer to call the parameter `xfail_strict` (same name as the pytest [config](https://docs.pytest.org/en/stable/reference/reference.html#confval-xfail_strict))?
```suggestion
    xfail_strict: bool | None = None,
```